### ISSUE TITLE
About the relative motion computation in EGs.txt

### ISSUE THREAD

**Issue by xgao-3dv** [wilsonkl/SfM_Init#19]
Hi Kyle,
As we know that relative rotation and translation is obtained by essential matrix decomposition, and intrinsic camera parameter is required for essential matrix estimation.
And my question is, what intrinsic camera parameter is used when getting the relative motion in EGs.txt? Is it simply the one recorded in the file of coords.txt as the item of 'focal'?
<img width="834" alt="coords" src="https://github.com/wilsonkl/SfM_Init/assets/121286720/7564e95b-9f88-4453-8830-312c13dd306e">
Thanks a lot
Xiang Gao

**Comment by wilsonkl** [comment]
Hi Xiang,

The person who is best able to answer your question is Noah Snavely. He was my dissertation advisor, and he still has access to the source material behind this data, and to the scripts that generated it. 

I can point you to this [focal length extraction script](https://github.com/snavely/bundler_sfm/blob/master/bin/extract_focal.pl) included with Bundler. It's how all of Noah's pipelines began focal length estimation. That estimate could sometimes change through bundle adjustment later, but focal lengths always started with that script.

This is reaching back a decade, but I remember that the in-house version of Bundler sometimes performed a bundle adjustment on a two-view model (and sometimes it skipped this step). That means (if I remember right) that some EGs reported in my dataset passed through an extra step of geometric refinement. I don't remember if focal length was frozen or free in that two-view bundle adjustment. I'd guess focal length was frozen because otherwise you'd get mismatched focal estimates for the same camera coming from different pairs. I don't remember seeing logic to deal with that. (Chris Sweeney later did a cool paper with the focal lengths of all two-view models jointly optimized together, but that came after my work.)

In short, I strongly suspect that there's only one focal length present in the pipeline at this early stage, and that it comes from Bundler's focal length perl script. So the coords.txt value should be the same one that is used for EG estimation.
